tests/test_poly.py

```diff
@@ -0,0 +1,287 @@
+
+    #We expect there to be these coefficients:
+    # x^0 y^0   1
+
+    # x^0 y^1   2
+    # x^1 y^0   3
+
+    # x^0 y^2   4
+    # x^1 y^1   5 
+    # x^2 y^0   6
+
+    # x^0 y^3   7
+    # x^1 y^2   8
+    # x^2 y^1   9
+    # x^3 y^0   10 
+
+    #Check that we have the indices we expect
+    assert interp.indices[0] == [
+        (0,0),
+        (0,1), (1,0),
+        (0,2), (1,1), (2,0),
+        (0,3), (1,2), (2,1), (3,0)
+    ]
+    assert interp.nvariables[0]==10
+
+    #check the packing then unpacking a 
+    packed = numpy.random.uniform(size=interp.nvariables[0])
+    unpacked = interp.unpack_coefficients(0,packed)
+    packed_test = interp.pack_coefficients(0,unpacked)
+
+    #Check that the shape is 4*4 in the unpacked (because we
+    #want space for all the terms), and that we can unpack and
+    #repack successfully.
+    assert (packed==packed_test).all()
+    assert unpacked.shape == (N+1,N+1)
+
+    unpacked_test = numpy.zeros_like(unpacked)
+
+    #check that we have zeros for the terms that should be zero in the matrix.
+    #We don't want any terms with total exponent > N.
+    for i in xrange(N+1):
+        for j in xrange(N+1):
+            if i+j>3:
+                assert unpacked[i,j]==0.0
+                unpacked_test[i,j]=0.0
+
+    #Now do the test the other way around, checking that 
+    #we can pack and then unpack
+    packed_test_2 = interp.pack_coefficients(0,unpacked_test)
+    unpacked_test_2 = interp.unpack_coefficients(0,packed_test_2)
+    assert (unpacked_test_2 == unpacked_test).all()
+
+def test_poly_mean():
+    #Zero'th order polynomial fitting should be pretty trivial, just
+    #the same as the mean fitting. So much of this code is just taken from
+    #the mean testing in test_simple
+    N = 0
+    nparam = 5
+    orders = [N for i in xrange(nparam)]
+    interp = piff.Polynomial(orders)
+    nstars = 100
+
+    #Choose some random values of star parameters
+    vectors = [ numpy.random.random(size=nparam) for i in range(nstars) ]
+
+    #take the mean of them. Our curve fit should be able to reproduce this.
+    mean = numpy.mean(vectors, axis=0)
+
+    #Choose some random positions in the field.
+    #NB think more about bounds and 
+    pos = [ (numpy.random.random()*10, numpy.random.random()*10)
+            for i in range(nstars) ]
+
+    # Run our solver.
+    interp.solve(pos, vectors)
+
+    #we expect one set of coefficients per object
+    assert len(interp.coeffs)==5
+
+    #We should have very close values (not necessarily identical) since
+    #we calculate these in numerically different ways.
+    for mu, val in zip(mean, interp.coeffs):
+        assert numpy.isclose(mu, val[0,0])
+
+    #We also expect that if we interpolate to any point we just
+    #get the mean as well
+    for i in xrange(30):
+        p=(numpy.random.random()*10, numpy.random.random()*10)
+        v = interp.interpolate(p)
+        assert numpy.allclose(v, mean)
+
+
+def sub_poly_linear(type1):
```

Comment:
I think nosetests will only run functions that start with (or maybe include?) `test`.  So should change the name of this function.